templates/dev.rake

```diff
@@ -1,4 +1,4 @@
-if Rails.env.development? || Rails.env.test?
+unless Rails.env.production?
```

Comment:
In my experience, the staging/production split is a source of potential mistakes: code doesn't run any place you test as it does in production. I don't have specifics, but remember unpleasant surprises. The idea makes me uncomfortable.

Makes sense to me to configure different services to different keys, and have the code run exactly the same.

In an app I needed to disable some services in staging because of costs, and I added a line similar to `ENV.fetch("HEROKU_APP_NAME", "").include?("staging-pr-")` to `Rails.application.config`. Can also suggest adding a boolean like `DEMO=true` or `STAGING_ENV=true`, that `HONEYBADGER_ENV` can mimic. This still rings bells in my brain, but it's been useful so far.

Development and test ought to run in different databases, use different mailer configurations, and sometimes fakes. I think it makes sense to configure those independently. The differences between the three `config/environments/*` files describe more examples.

Thanks for your discussion. I've been taking this for granted for too long. :)